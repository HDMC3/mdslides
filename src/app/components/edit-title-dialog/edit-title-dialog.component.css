form {
    display: flex;
    flex-direction: column;
    gap: 0.5rem;
}

.dialog-footer {
    display: flex;
    justify-content: space-evenly;
    align-items: center;
}