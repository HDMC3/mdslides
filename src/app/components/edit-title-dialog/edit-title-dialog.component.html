<nb-card>
    <nb-card-header>
        <h4>
            Editar titulo
        </h4>
    </nb-card-header>
    <nb-card-body>
        <form [formGroup]="titleForm">
            <label for="title">Titulo:</label>
            <input formControlName="title" id="title" nbInput type="text" [status]="titleControlStatus" autocomplete="off">
            <small *ngIf="!titleForm.controls['title'].valid" style="color: var(--color-danger-500)">El titulo es requerido</small>
        </form>
    </nb-card-body>
    <nb-card-footer>
        <div class="dialog-footer">
            <button nbButton status="danger" (click)="closeDialog()">Cancelar</button>
            <button nbButton status="success" (click)="save()">Guardar</button>
        </div>
    </nb-card-footer>
</nb-card>