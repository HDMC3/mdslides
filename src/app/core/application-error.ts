export class ApplicationError extends Error {
    constructor(message: string, public title?: string) {
        super(message);
    }
}
