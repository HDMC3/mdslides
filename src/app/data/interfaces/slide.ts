export interface Slide {
    id: string;
    code: string;
    name: string;
}
