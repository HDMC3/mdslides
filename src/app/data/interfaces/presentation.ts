import { Slide } from './slide';

export interface Presentation {
    id: string;
    title: string;
    description: string;
    slides: Slide[];
    creation_date: Date;
    modification_date: Date;
}
