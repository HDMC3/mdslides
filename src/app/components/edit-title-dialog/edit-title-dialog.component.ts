import { Component, Input, OnDestroy, OnInit } from '@angular/core';
import { FormControl, FormGroup, Validators } from '@angular/forms';
import { NbComponentOrCustomStatus, NbDialogRef } from '@nebular/theme';
import { Subscription } from 'rxjs';
import { FormValidators } from 'src/app/shared/form-validators';
import { Presentation } from 'src/app/data/interfaces/presentation';

@Component({
    selector: 'app-edit-title-dialog',
    templateUrl: './edit-title-dialog.component.html',
    styleUrls: ['./edit-title-dialog.component.css']
})
export class EditTitleDialogComponent implements OnInit, OnDestroy {

    @Input() presentation!: Presentation;
    titleForm: FormGroup;
    titleControlStatus: NbComponentOrCustomStatus = 'basic';
    formStatusChanges: Subscription;

    constructor(private nbDialogRef: NbDialogRef<any>) {
        this.titleForm = new FormGroup({
            title: new FormControl('', [Validators.required, FormValidators.noEmpty])
        });

        this.formStatusChanges = this.titleForm.statusChanges.subscribe(status => {
            this.titleControlStatus = status === 'INVALID' ? 'danger' : 'basic';
        });
    }

    ngOnInit(): void {
        this.titleForm.controls['title'].setValue(this.presentation.title);
    }

    ngOnDestroy(): void {
        this.formStatusChanges.unsubscribe();
    }

    save() {
        if (this.titleForm.invalid) {
            return;
        }

        this.presentation.title = this.titleForm.controls['title'].value.trim();
        this.closeDialog();
    }

    closeDialog() {
        this.nbDialogRef.close();
    }

}
