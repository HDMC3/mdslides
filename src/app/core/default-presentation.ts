import { v4 } from 'uuid';
import { Presentation } from '../data/interfaces/presentation';

export const getDefaultPresentation = (id: string): Presentation => {
    return {
        id,
        title: 'Titulo de presentacion',
        creation_date: new Date(),
        modification_date: new Date(),
        description: '',
        slides: [
            {
                id: v4(),
                name: 'Presentacion inicial',
                code: [
                    '---',
                    'style: |',
                    'section {',
                    '   display: flex;',
                    '   flex-direction: column;',
                    '   align-items: center;',
                    '   justify-content: center;',
                    '}',
                    '---',
                    '',
                    '# Inicio de la presentacion',
                    '# ðŸ˜„âœŒ'
                ]
            }
        ]
    };
};
