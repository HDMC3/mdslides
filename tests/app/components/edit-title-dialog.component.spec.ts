import { ComponentFixture, TestBed } from '@angular/core/testing';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { NbEvaIconsModule } from '@nebular/eva-icons';
import { NbButtonModule, NbCardModule, NbDialogModule, NbDialogRef, NbIconModule, NbInputModule, NbThemeModule } from '@nebular/theme';
import { Observable } from 'rxjs';
import { PresentationService } from 'src/app/core/services/presentation.service';
import { Presentation } from 'src/app/data/interfaces/presentation';

import { EditTitleDialogComponent } from '../../../src/app/components/edit-title-dialog/edit-title-dialog.component';

describe('EditTitleDialogComponent', () => {
    let component: EditTitleDialogComponent;
    let fixture: ComponentFixture<EditTitleDialogComponent>;
    let compiled: HTMLElement;

    const inputPresentation: Presentation = {
        id: '',
        creation_date: new Date(),
        title: 'Titulo de presentacion',
        slides: []
    };

    const nbDialogRef = {
        close() { },
        componentRef: {},
        onBackdropClick: new Observable<MouseEvent>(),
        onClose: new Observable()
    };

    beforeEach(async() => {
        await TestBed.configureTestingModule({
            declarations: [EditTitleDialogComponent],
            imports: [
                FormsModule,
                ReactiveFormsModule,
                NbThemeModule.forRoot({ name: 'default' }),
                NbCardModule,
                NbIconModule,
                NbEvaIconsModule,
                NbButtonModule,
                NbInputModule,
                NbDialogModule.forRoot()
            ],
            providers: [
                PresentationService,
                { provide: NbDialogRef, useValue: nbDialogRef }
            ]
        })
            .compileComponents();
    });

    beforeEach(() => {
        fixture = TestBed.createComponent(EditTitleDialogComponent);
        component = fixture.componentInstance;
        component.presentation = inputPresentation;
        fixture.detectChanges();
        compiled = fixture.nativeElement;
    });

    it('should create', () => {
        expect(component).toBeTruthy();
    });

    it('should match snapshot', () => {
        expect(compiled).toMatchSnapshot();
    });

    it('presentation title value should not change if title form is invalid', () => {
        const beforeTitleValue = component.presentation.title;
        component.titleForm.controls['title'].setValue('');

        const footerButtons = compiled.querySelectorAll<HTMLButtonElement>('.dialog-footer button');
        const saveButton = footerButtons[1];
        saveButton.click();

        expect(beforeTitleValue).toBe(component.presentation.title);
    });
});
