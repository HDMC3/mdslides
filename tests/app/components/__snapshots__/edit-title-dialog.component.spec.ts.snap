// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditTitleDialogComponent should match snapshot 1`] = `
<div
  id="root1"
>
  <nb-card>
    <nb-card-header>
      <h4>
         Editar titulo 
      </h4>
    </nb-card-header>
    <nb-card-body>
      <form
        class="ng-untouched ng-pristine ng-valid"
        novalidate=""
      >
        <label
          for="title"
        >
          Titulo:
        </label>
        <input
          autocomplete="off"
          class="ng-untouched ng-pristine ng-valid size-medium status-basic shape-rectangle"
          formcontrolname="title"
          id="title"
          nbinput=""
          type="text"
        />
        
      </form>
    </nb-card-body>
    <nb-card-footer>
      <div
        class="dialog-footer"
      >
        <button
          aria-disabled="false"
          class="appearance-filled size-medium shape-rectangle status-danger"
          nbbutton=""
          status="danger"
          tabindex="0"
        >
          Cancelar
        </button>
        <button
          aria-disabled="false"
          class="status-success appearance-filled size-medium shape-rectangle"
          nbbutton=""
          status="success"
          tabindex="0"
        >
          Guardar
        </button>
      </div>
    </nb-card-footer>
  </nb-card>
</div>
`;
